### ISSUE TITLE
'Ignore' setting doesn't work

### ISSUE THREAD

**Issue by michalsnik** [sasstools/sass-lint#452]
Hi,

Straight to the point.
I created simple files structure:
<img width="157" alt="zrzut ekranu 2015-12-13 23 34 29" src="https://cloud.githubusercontent.com/assets/4093559/11769904/9e14d38a-a1f2-11e5-8888-e969ae43a36e.png">

Then I copied .sass-lint.yml from here: https://github.com/sasstools/sass-lint/blob/develop/lib/config/sass-lint.yml

And modified it by adding 'ignore' setting:
<img width="203" alt="zrzut ekranu 2015-12-13 23 34 50" src="https://cloud.githubusercontent.com/assets/4093559/11769919/d5708478-a1f2-11e5-95bb-463fa2b0c6a4.png">

My gulpfile looks like this:
<img width="310" alt="zrzut ekranu 2015-12-13 23 40 45" src="https://cloud.githubusercontent.com/assets/4093559/11769933/f50893fc-a1f2-11e5-81ec-0b1dc4c76cf7.png">

When i run gulp it lints every .scss file in my project, and it doesn't ignore files I set in 'ignore' setting.

**Comment by michalsnik** [comment]
Fixed in #451

**Comment by DanPurdy** [comment]
Sorry @michalsnik, I will look into this just incredibly busy times at the moment. Hopefully by the end of the week I'll have some answers for you.

**Comment by DanPurdy** [comment]
As I mention in the PR I believe this is actually a `gulp-sass-lint issue` as it's not passing through our `lintFiles` method to handle ignores and includes properly at the moment. The behaviour intended above works perfectly fine from the CLI and other integrations.

**Comment by sbalay** [comment]
@DanPurdy 
Is there any action planned to solve this? Whether its on sass-lint or on gulp-sass-lint?

**Comment by DanPurdy** [comment]
@sbalay, yes there's a need to rethink how our config file is loaded, I believe we're looking to sort this in sass-lint itself but at the moment there are a few PR's of mine waiting to be merged which affect the functionality that would need to change so i want to see those merged first, unfortunately all of the contributors including myself haven't really had the time to work on this recently. Very soon though hopefully...

**Comment by cadriel** [comment]
We've just hit this too - I was wondering why the `include` property worked in CLI but not my gulp task. This is mostly an issue for us because none of the partials are checked.

**Comment by DanPurdy** [comment]
It's the method that gulp-sass-lint attaches itself to doesn't do any of the config parsing with regards to include and excluded files from your config. Basically we need to update this or the gulp plugin itself would need to parse your config too which isn't ideal..

I intend to get something working here for 1.6 once the last rule update is out of the way..

**Comment by webgefrickel** [comment]
I am using the gulp-plugin and I still can reproduce this issue with 1.6.0. Is this something that should be done in the gulp-plugin or in here?

**Comment by DanPurdy** [comment]
It's unfortunately one of those that could either be sass-lint or gulp-sass-lint.

I intend to look into this and handle it with sass-lint soon though, just need some time now to get a working solution together without forcing a major release/update number

**Comment by webgefrickel** [comment]
For now I am using a workaround in gulp like this:

```javascript
...
    return gulp
      .src([
        'src/scss/**/*.scss',
        '!src/scss/vendor/*.scss'
      ])
      .pipe(sassLint())
      .pipe(sassLint.format());
...
```

Maybe this helps other people as well...

**Comment by DanPurdy** [comment]
Yeah, essentially you've hit the nail on the head there, It's just because we dont check the config in the file that the gulp plugin attaches itself to as it's been checked further up the usual chain. We need to rectify that.

**Comment by DanPurdy** [comment]
Sorry for how long this has been hanging about, with the next release of sass-lint and the next release of gulp-sass-lint the ignores in your config will be read and respected again. Includes still won't as they should be passed as glob patterns to gulp-src anyway.
